After pinning the dependencies in conda envs I was able to get past rule `postproc_boundary_vertices` but now it seems to be failing at rule `get_boundary_vertices`. I'm running the following command:
```
/hippunfold/run.py ../lowresMRI/ ../test-lowresMRI/ participant --participant-label 01 --use-conda --modality T1w --cores all
``` 

Error from the log file:
```
2025-04-24 12:55:50,684 - INFO - Loading surface from GIFTI...
Traceback (most recent call last):
  File "/local/scratch/test-lowresMRI/.snakemake/scripts/tmpmit2ffk1.get_boundary_vertices.py", line 15, in <module>
    surface, metadata = read_surface_from_gifti(snakemake.input.surf_gii)
  File "/local/scratch/hippunfold/hippunfold/workflow/lib/surface.py", line 26, in read_surface_from_gifti
    metadata = dict(vertices_darray.meta) if vertices_darray else {}
TypeError: 'GiftiMetaData' object is not iterable

```

--- a/hippunfold/workflow/envs/pyvista.yaml
+++ b/hippunfold/workflow/envs/pyvista.yaml
@@ -5,7 +5,7 @@ channels:
 dependencies:
   - python=3.9
   - numpy>=1.21.2,<2.0.0
-  - nibabel>=3.2.1,<4.0.0
+  - nibabel>=3.2.1
   - pyvista>=0.44.2,<0.45.0
   - vtk
   - scipy

--- a/hippunfold/workflow/scripts/gen_isosurface.py
+++ b/hippunfold/workflow/scripts/gen_isosurface.py
@@ -56,14 +56,12 @@ coords[sink_mask == 1] = 1.1
 src_sink_nan_mask = get_adjacent_voxels(sink_mask, src_mask)
 coords[src_sink_nan_mask == 1] = np.nan
 
-
 grid.cell_data["values"] = coords.flatten(order="F")
 grid = grid.cells_to_points("values")
-tfm_grid = grid.transform(affine, inplace=False)
 
 # Generate isosurface
 logger.info("Generating isosurface")
-surface = tfm_grid.contour(
+surface = grid.contour(
     [snakemake.params.threshold], method="contour", compute_scalars=True
 )
 logger.info(surface)
@@ -93,6 +91,9 @@ logger.info(surface)
 
 logger.info(f"final surface clean to remove overlapping vertices, etc.")
 surface = surface.clean()
+
+# apply affine to go from matrix to image space
+surface = surface.transform(affine, inplace=False)
 
 # Save the final mesh
 write_surface_to_gifti(surface, snakemake.output.surf_gii)

